### PR TITLE
DOC-ONLY-install KKM using kkm.run consistently

### PR DESCRIPTION
**Pull request by smijar**

### CODE REVIEW

**Review by gnode1 — COMMENTED** [review]
Should we fix kkm.run to take these steps instead?

**Comment by smijar** [comment]
That would be ideal yes, I think.  I did notice that kkm.run was downloading kernel-version-16-devel on a kernel version 10 system.

**Comment by gnode1** [comment]
> Should we fix kkm.run to take these steps instead?

Yes, we should fix the install script so it will take care of that.

**Comment by smijar** [comment]
So to do this we have to update [lines](https://github.com/kontainapp/kkm/blob/32970b6dbc889c8d32a2049ed6f9546d8b8bf62c/installer/build-script.sh#L55-L68) 

```shell
# for dnf we have to include: 
$ dnf update --exclude=kernel*

# for yum, as well:
$ yum update --exclude=kernel*

# for apt luckily it looks 'apt update' does not update kernel, but to be sure we can add this, but needs a little testing: 
$ apt-mark hold linux-image-generic linux-headers-generic
```

File: docs/kkm_install_amznlinux2_fedora [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+yum -y --exclude=kernel* update
+
+# check if it needs rebooting while yum not running
+(test ! -f /var/run/yum.pid && needs-restarting -r) || (reboot)
+```
+
+For Fedora:
+```shell
+# install kernel-devel for current package
+sudo dnf install -y "kernel-devel-uname-r == $(uname -r)" dnf-utils
```
Comment (gnode1) [inline docs/kkm_install_amznlinux2_fedora:22]: `sudo` here is unnecessary - `dnf install` requires root, so does reboot.

File: docs/kkm_install_amznlinux2_fedora [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+This prevents errors that pop up during the build with header locations when building KKM on target system.
+
+For Amazon Linux 2:
+```shell
+yum install -y "kernel-devel-uname-r == $(uname -r)" yum-utils
+# update without updating kernel (IMPORTANT FOR KKM)
+yum -y --exclude=kernel* update
+
+# check if it needs rebooting while yum not running
+(test ! -f /var/run/yum.pid && needs-restarting -r) || (reboot)
```
Comment (gnode1) [inline docs/kkm_install_amznlinux2_fedora:16]: ASAIK the parenthesis are not necessary. Also, we are not rebooting if yum is running. What happens then?

**Pull request closed without merging**